code/modules/mob/living/simple_animal/guardian/types/assassin.dm

```diff
@@ -1,100 +1,83 @@
-//Assassin
-/mob/living/simple_animal/hostile/guardian/assassin
-	melee_damage_lower = 15
-	melee_damage_upper = 15
-	attacktext = "slashes"
-	attack_sound = 'sound/weapons/bladeslice.ogg'
-	damage_coeff = list(BRUTE = 1, BURN = 1, TOX = 1, CLONE = 1, STAMINA = 0, OXY = 1)
-	playstyle_string = "<span class='holoparasite'>As an <b>assassin</b> type you do medium damage and have no damage resistance, but can enter stealth, massively increasing the damage of your next attack and causing it to ignore armor. Stealth is broken when you attack or take damage.</span>"
-	magic_fluff_string = "<span class='holoparasite'>..And draw the Space Ninja, a lethal, invisible assassin.</span>"
-	tech_fluff_string = "<span class='holoparasite'>Boot sequence complete. Assassin modules loaded. Holoparasite swarm online.</span>"
-	carp_fluff_string = "<span class='holoparasite'>CARP CARP CARP! Caught one! It's an assassin carp! Just when you thought it was safe to go back to the water... which is unhelpful, because we're in space.</span>"
+//ass ass in
 
-	toggle_button_type = /obj/screen/guardian/ToggleMode/Assassin
-	var/toggle = FALSE
+/datum/guardian_abilities/assassin
+	id = "assassin"
+	name = "Undetected Elimination"
+	value = 6
 	var/stealthcooldown = 160
 	var/obj/screen/alert/canstealthalert
 	var/obj/screen/alert/instealthalert
 
-/mob/living/simple_animal/hostile/guardian/assassin/Initialize()
-	..()
-	stealthcooldown = 0
+/datum/guardian_abilities/assassin/Destroy()
+	return ..()
 
-/mob/living/simple_animal/hostile/guardian/assassin/Life()
-	. = ..()
-	updatestealthalert()
-	if(loc == summoner && toggle)
-		ToggleMode(0)
+/datum/guardian_abilities/assassin/handle_stats()
+	guardian.has_mode = TRUE
```

Comment:
@ChangelingRain what if it's something that's kind of compatible like assassin and ranged?